bioimageio/core/measure_groups.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import warnings
+from itertools import product
+from typing import Dict, Hashable, Iterator, List, Mapping, Optional, Sequence, Tuple, Type, Union
+
+import numpy
+import xarray as xr
+
+from bioimageio.core.prediction_pipeline import TensorName
+from bioimageio.core.statistical_measures import Mean, Measure, Percentile, Std, Var
+
+try:
+    import crick
+except ImportError:
+    crick = None
+
+MeasureValue = xr.DataArray
+
+
+class MeasureGroup:
+    def reset(self):
+        raise NotImplementedError
+
+    def update_with_sample(self, sample: Dict[TensorName, xr.DataArray]):
+        raise NotImplementedError
+
+    def finalize(self) -> Dict[Measure, MeasureValue]:
+        raise NotImplementedError
+
+
+class MeanVarStd(MeasureGroup):
+    n: int
+    mean: Optional[xr.DataArray]
+    m2: Optional[xr.DataArray]
+
+    def __init__(self, tensor_name: TensorName, axes: Optional[Tuple[int]]):
+        self.axes: Optional[Tuple[str]] = axes
+        self.tensor_name = tensor_name
+        self.reset()
+
+    def reset(self):
+        self.n = 0
+        self.mean = None
+        self.m2 = None
+
+    def update_with_sample(self, sample: Dict[TensorName, xr.DataArray]):
+        tensor = sample[self.tensor_name]
+        mean_b = tensor.mean(dim=self.axes)
+        n_b = numpy.prod(tensor.shape) / numpy.prod(mean_b.shape)  # reduced voxel count
```

Comment:
@constantinpape I'm not sure with the count/weight. Simply taking count 1 for 1 sample might neglect that samples may differ in the reduced dimensions. Thus here I take the voxel count as weight, but these may be big numbers, so I worry about numerical stability (see update formulas below)

Reply:
I am now casting the input tensor to flaot64 and added a few asserts to ensure it stays there. For the percentiles I store the percentiles as float64 in `MeanPercentiles` although the error is probably big here anyway.... https://github.com/bioimage-io/core-bioimage-io-python/commit/a7c920669831190fde3c58b79746ccf3cd688f4f